### ISSUE TITLE
How to check if a field is validated

### ISSUE THREAD

**Issue by yordis** [react-hook-form/react-hook-form#3457]
**Is your feature request related to a problem? Please describe.**
https://github.com/react-hook-form/react-hook-form/discussions/2633

**Describe the solution you'd like**
https://github.com/react-hook-form/react-hook-form/discussions/2633

**Describe alternatives you've considered**

Copy-paste  the code:

```tsx
export function isFieldValid<T extends FieldValues>(field: keyof T, form?: UseFormMethods<T>) {
  return form.formState?.dirtyFields[field] && !form.formState?.errors[field];
}
```

Example usage:

```tsx
import { useFormContext } from 'react-hook-form';
import { getFieldState } from '@/helpers/react-hook-form';
import { InputField } from '@/components/forms/input-field';
import { EMAIL_REGEX } from '@/constants';

export function EmailInput() {
  const form = useFormContext();
  const state = getFieldState('email', form);
  return (
    <div className="flex flex-col">
      <InputField
        name="email"
        label="Email"
        type="email"
        inputMode="email"
        placeholder="Your email"
        autoFocus
        autoComplete="email"
        tabIndex={1}
        state={state}
        ref={form.register({
          required: 'Your email is required',
          pattern: {
            value: EMAIL_REGEX,
            message: 'Your email is not a valid email',
          },
        })}
      />
    </div>
  );
}
```

**Comment by bluebill1049** [comment]
Thanks 👍 will keep an eye on this and pinged to the board.

**Comment by bluebill1049** [comment]
related https://github.com/react-hook-form/react-hook-form/issues/3434

**Comment by yordis** [comment]
@bluebill1049 wouldn't be better to keep the issue open? Where do I suppose to subscribe to changes related to this?

**Comment by bluebill1049** [comment]
I keep a list of all feature requests open here: https://github.com/react-hook-form/react-hook-form/issues/1981 issues are for tracking actual issues. you can view them there: https://github.com/react-hook-form/react-hook-form/projects/1

**Comment by yordis** [comment]
Personally, I don't think that most people will understand the way you manage issues, also, I don't the way to upvote, I am a bit confused about what I suppose to do thou.

I would recommend leveraging Projects and Issues at the same time because it is hard to follow the project management done this way.

That being said, it is your project, and I am grateful for your amazing work regardless of how you are handling the at the moment.

**Comment by bluebill1049** [comment]
Thanks, @yordis for your feedback and understanding. I do ping the feature request at the issues board. 

<img width="618" alt="Screen Shot 2020-11-17 at 9 38 35 am" src="https://user-images.githubusercontent.com/10513364/99316546-ac047980-28b8-11eb-924d-42dbbcdac0aa.png">

Honestly, I am one man who tries to maintain the project here, so this way works for me over the year, so I can focus on the issues and keep away those features in separate sections instead of issues. Hope you understand 🙏

**Comment by bluebill1049** [comment]
proposal: 

```tsx
<Controller 
  render={(props, { isDirty, isValid, error, isTouched } ) => {
    console.log(state)
  }}
/>
```

**Comment by yordis** [comment]
LETS GOOO

**Comment by iDuuck** [comment]
Would love to get this done in an PR. However, I am not sure if this is already being implemented, since it is on “In Progress”.

Also the proposed API is not working for me, since “state” is not defined?

**Comment by bluebill1049** [comment]
@iDuuck it's in RFC: https://github.com/react-hook-form/react-hook-form/discussions/3487
